### PR TITLE
[FLINK-17865][checkpoint] Increase default size of 'state.backend.fs.memory-threshold'

### PR DESCRIPTION
**Pull request by Myasuka**
## What is the purpose of the change

Increase the default value of `state.backend.fs.memory-threshold` from `1K` to `20K` as [thread](https://s.apache.org/us0en) discussed.

## Brief change log

  - Increase default size of 'state.backend.fs.memory-threshold' from `1k` to `20k`

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no 
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: yes
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature?  no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by Myasuka** [comment]
My azure CI pipeline has run successfully twice: [my-build-1](https://myasuka.visualstudio.com/flink/_build/results?buildId=77) and [my-build-2](https://myasuka.visualstudio.com/flink/_build/results?buildId=75)

Current CI in this PR failed due to FLINK-16572

@flinkbot run azure

File: flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendLoadingTest.java [hunk 1]
```diff
@@ -227,7 +228,7 @@ public void testLoadFileSystemStateBackend() throws Exception {
 		final String savepointDir = new Path(tmp.newFolder().toURI()).toString();
 		final Path expectedCheckpointsPath = new Path(checkpointDir);
 		final Path expectedSavepointsPath = new Path(savepointDir);
-		final int threshold = 1000000;
+		final MemorySize threshold = MemorySize.parse("900kb");
```
Comment (klion26) [inline flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendLoadingTest.java:231]: Do you think we need to add a test case to verify the default value of `state.backend.fs.memory-threshold`(do not set this key, and verify the result of `getMinFileSizeThreshold()` is the expected value)

**Comment by Myasuka** [comment]
Another successful run in my local branch: [my-build](https://dev.azure.com/myasuka/flink/_build/results?buildId=78&view=results)

**Comment by Myasuka** [comment]
@flinkbot run azure

File: flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsStateBackend.java [hunk 2]
```diff
@@ -388,6 +388,10 @@ private FsStateBackend(FsStateBackend original, ReadableConfig configuration, Cl
 		this.writeBufferSize = Math.max(bufferSize, this.fileStateThreshold);
 	}
 
+	private boolean checkFileSateThresholdValid(int fileStateThreshold) {
```
Comment (StephanEwen) [inline flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsStateBackend.java:391]: Typo in this method name.

File: flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendLoadingTest.java (hunk 1 above)
Comment (Myasuka) [reply flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendLoadingTest.java:231]: I think current [test_get_min_file_size_threshold](https://github.com/apache/flink/blob/e92b2bf19bdf03ad3bae906dc5fa3781aeddb3ee/flink-python/pyflink/datastream/tests/test_state_backend.py#L100) should already verify it.

**Comment by StephanEwen** [comment]
Looks good to me, +1 to merge this

**Comment by carp84** [comment]
The e2e tests (#12373) for release-1.11 passed thus I will merge the changes. Thanks.

**Pull request closed without merging**